guides/examples/Apache Kafka.md

```diff
@@ -0,0 +1,214 @@
+initializing Kafka, you can crate a new stream by running:
+
+    kafka-topics --create --zookeeper localhost:2181 --partitions 3 --topic test
+
+## Configure your Elixir project to use Broadway
+
+This guide describes the steps necessary to integrate Broadway with Kafka using
+[BroadwayKafka](https://github.com/plataformatec/broadway_kafka),
+which is a Broadway Kafka Connector provided by [Plataformatec](http://www.plataformatec.com).
+
+BroadwayKafka can subscribe to one or more topics and process streams of records
+using Kafka's [Consumer API](https://kafka.apache.org/documentation.html#consumerapi).
+
+Each GenStage producer initialized by BroadwayKafka will be available as a consumer,
+all registered using the same self-labed **consumer group**. Each record published to a
+topic/partition will be delivered to one consumer instance within each consumer group.
+
+Bear in mind that a topic/partition can be assigned to any consumer instance that has
+been subscribed using the same consumer group, i.e, any Broadway instance or application
+running on any machine connected to the Kafka cluster.
+
+### Starting a new project
+
+Create a new project running:
+
+    mix new my_app --sup
+
+The `--sup` flag instructs Elixir to generate an application with a supervision tree.
+
+### Setting up dependencies
+
+Add `:broadway_kafka` to the list of dependencies in `mix.exs`:
+
+    def deps do
+      [
+        ...
+        {:broadway_kafka, "~> 0.1.0"}
+      ]
+    end
+
+Don't forget to check for the latest version of dependencies.
+
+## Define the pipeline configuration
+
+Broadway is a process-based behaviour and to define a Broadway pipeline,
+we need to define three functions: `start_link/1`, `handle_message/3`
+and optionally `handle_batch/4`. We will cover `start_link/1` in this
+section and the `handle_` callbacks in the next one.
+
+Similar to other process-based behaviours, `start_link/1` simply
+delegates to `Broadway.start_link/2`, which should define the
+producers, processors, and batchers in the Broadway pipeline.
+Assuming we want to consume messages from a queue called
+`my_queue`, one possible configuration would be:
+
+    defmodule MyBroadway do
+      use Broadway
+
+      alias Broadway.Message
+
+      def start_link(_opts) do
+        Broadway.start_link(__MODULE__,
+          name: __MODULE__,
+          producer: [
+            module: {BroadwayKafka.Producer, [
+              hosts: [localhost: 9092],
+              group_id: "group_1",
+              topics: ["test"],
+            ]},
+            stages: 10
+          ],
+          processors: [
+            default: [
+              stages: 10
+            ]
+          ],
+          batchers: [
+            default: [
+              batch_size: 100,
+              batch_timeout: 200,
+              stages: 10
+            ]
+          ]
+        )
+      end
+
+      ...callbacks...
+    end
+
```

Comment:
It is worth adding a note here that says: "Pipelines built on top of BroadwayKafka are automatically partitioned. So even though there are multiple processes, these processes will preserve Kafka's ordering semantics when it comes to topics/partitions"?